### PR TITLE
added mantine and setup some layouts

### PR DESCRIPTION
**Pull request by NatoNathan**

### CODE REVIEW

**Comment by MiniCheese26** [comment]
There's a [`use-color-scheme`](https://mantine.dev/hooks/use-color-scheme/) hook for getting the preferred colour scheme

**Comment by NatoNathan** [comment]
Cool, we should probably check that then allow users to override it with the button? On Sat, Sep 17, 2022 at 16:49, Emily Medhurst ***@***.***> wrote:  
There's a use-color-scheme hook for getting the preferred colour scheme

**Comment by MiniCheese26** [comment]
Yep that's what I was thinking. And store the override in either local storage or a cookie as the docs suggest.

**Comment by NatoNathan** [comment]
Ok, I will change this when I get home.

File: src/hooks/useColorMode.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    getInitialValueInEffect: true,
+  });
+
+  const toggleColorScheme = (value?: ColorScheme|'preferredColorScheme') => setColorScheme(value || (colorScheme === 'dark' ? 'light' : 'dark'));
+
+  useHotkeys([['mod+J', () => toggleColorScheme()]]);
+  useHotkeys([['mod+shift+J', () => toggleColorScheme('preferredColorScheme')]]);
+
+  return { colorMode: colorScheme !== 'preferredColorScheme'? colorScheme: preferredColorScheme, toggleColorMode: toggleColorScheme };
+};
```
Comment (NatoNathan) [inline src/hooks/useColorMode.ts:18]: this is the new color mode, I think this works well. The hotkey can be changes if needed tho

Comment (NatoNathan) [reply src/hooks/useColorMode.ts:18]: this way it stays in sync with the device

Comment (MiniCheese26) [reply src/hooks/useColorMode.ts:18]: Looks good

**Pull request merged**